src/ReflectionClosure.php

```diff
@@ -305,6 +306,7 @@ public function getCode()
                         case ']':
                             if ($isShortClosure) {
                                 if ($open === 0) {
+                                    $semicolonNeeded = $token[0] === ')' && $tokens[$i+1] === ';';
```

Comment:
I believe that this line is not needed at all. Am I missing something?

Reply:
Why not? 
The first case `(fn () => 1)` is interpreting as `fn () => 1` (*brackets are removed*), because this boundary condition is required.

If you know how to fix it welcome ;)

Reply:
Also we no need to add semicolon if function is in array